shaka/src/js/dom/document.cc

```diff
@@ -64,6 +65,14 @@ RefPtr<Element> Document::DocumentElement() const {
   return nullptr;
 }
 
+RefPtr<Attr> Document::CreateAttribute(const std::string& name) {
+  return new Attr(this, nullptr, util::ToAsciiLower(name), nullopt, nullopt, "");
+}
+
+RefPtr<Attr> Document::CreateAttributeNS(const std::string& namespace_uri, const std::string& name) {
+  return new Attr(this, nullptr, name, namespace_uri, nullopt, "");
```

Comment:
`name` here could refers to a namespace qualified name (e.g. `foo:attr`).  You'd need to split on `:` and use the first value as the namespace prefix and the second as the local name. See `Element::SetAttributeNS` for something similar (although here it can be a simple name).

Reply:
Oh interesting! I must have overlooked this because `createAttribute` doesn't split but `createAttributeNS` does. Woops. I'll see about updating this a little later.